var/spack/repos/builtin/packages/npb/package.py

```diff
@@ -88,23 +117,23 @@ class Npb(MakefilePackage):
 
     @property
     def build_directory(self):
-        if '+mpi' in self.spec:
+        if 'implementation=mpi' in self.spec:
             implementation = 'MPI'
-        elif '+openmp' in self.spec:
+        elif 'implementation=openmp' in self.spec:
             implementation = 'OMP'
-        elif '+serial' in self.spec:
+        elif 'implementation=serial' in self.spec:
             implementation = 'SER'
         else:
             raise RuntimeError('You must choose an implementation to build')
```

Comment:
You can get rid of this `RuntimeError` now that it is physically impossible to specify an invalid implementation. That's my favorite features of multi-valued variants.